### DISCUSSION TITLE
Does anyone know of a list of triggers?

### DISCUSSION THREAD

**Discussion by cliveportman** [supabase/supabase#8814] [Questions] [answered]
Like `on_auth_user_created` but I'm looking for `on_auth_user_confirmed` or something similar.

Because of the max 24 hours for email confirmations, I'm having to deliver a 'resend invite' button for the client's admin team to use. But I want to add a column in the profile table called 'confirmed' or something similar that's filled when a user has confirmed. At the moment, I have nothing to test against so have these permanent 'resend invite' buttons.

**Comment by GaryAustin1** [comment, marked as answer]
I've not seen a list.  Triggers are just calls to a function to do what ever you want.
So if you want to change your profile table when the auth.user table is updated you would do a after update trigger and in that function check for what is being updated.   Updating the email column with a value would indicate confirmation of the email then you would update your profile table as needed.

Here is an example (last post) that is claimed to work.

https://github.com/supabase/supabase/discussions/5421

**Reply by cliveportman** [reply]
I was hoping there'd be an event I could listen for, but I keep forgetting Supabase doesn't work like that. The email column is already populated as the user has been invited already. Instead, I could listen for a user being updated and check the `Last Sign In` and look for a date that matches today and then update a column in the profiles table. That should work. It just means I'm listening for every user update rather than a one-off event per user, which seems a bit dirty to me.

**Reply by GaryAustin1** [reply]
There are no events (at least in the sense you are talking about) in Postgres besides triggers on table operations and realtime (which is not useful here).

**Reply by cliveportman** [reply]
Actually, do you know if when you listen for a table update, you're notified before or after the update? I want to check for an empty `Last Sign In` column but if the trigger is after the update, then I won't have the old value.

**Reply by GaryAustin1** [reply]
You choose BEFOrE or AFTER as part of the trigger definition.  An AFTER UPDATE trigger call to function still gets the old value. You just can not change the new value. 

https://www.postgresql.org/docs/11/plpgsql-trigger.html

**Reply by juanvilladev** [reply]
You can run:
`SELECT  event_object_table AS table_name ,trigger_name         
FROM information_schema.triggers  
GROUP BY table_name , trigger_name 
ORDER BY table_name ,trigger_name `
It'll list your triggers.

**Reply by GaryAustin1** [reply]
> You can run: `SELECT event_object_table AS table_name ,trigger_name FROM information_schema.triggers GROUP BY table_name , trigger_name ORDER BY table_name ,trigger_name ` It'll list your triggers.

The OP was looking for built in "triggers" (really hooks) on Auth events, not "Postgres" triggers on a table.
